### PR TITLE
Run3-sim47 Remove reference to trackermaterial in materials.xml

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Several cfi files use only materials.xml and not trackermaterial.xml. A couple of materials refer to trackermaterial.xml for elements which are identically defined in materials.xml. Removing the references will allow the cfi's which only used materials.xml to work without any warnings.

#### PR validation:

Tested with runTheMatrix.py for standard workflows

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by ianna** [comment]
-1

@bsunanda - we cannot change legacy scenarios. In addition, the materials used in Tracker should be moved out of **materials.xml** so that they can still use and modify **T_Copper** and **T_Aluminum** for zero material studies. Please see the issue https://github.com/cms-sw/cmssw/issues/28285

**Comment by ianna** [comment]
> @ianna I am happy as long as materials.xml does not refer to trackermaterial

One should move these materials from **materials.xml** to **trackermaterial.xml** and update all Tracker xml files that use them. Then update the payloads.

Alternatively, you can use the new **materials.xml** version since it's not used in the payloads and can be modified:
```
Geometry/CMSCommonData/data/materials/2021/v1/materials.xml
```

**Comment by bsunanda** [comment]
@ianna I shall modify the xml files to do that but can someone update the payloads. But I don't see why payloads to be created. Moving the definition from one file to another will not change the physical quantities

**Comment by bsunanda** [comment]
@ianna I updated all trackermaterial.xml and materials.xml files.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
+upgrade

**Comment by ianna** [comment]
-1

Material variation scenarios have to use T_* materials.

**Review by ianna — CHANGES_REQUESTED** [review]
@bsunanda, please, check the comments and cleanup the xml files and their scenarios. Most of the material variation definitions were produced automatically. Modifying them by hand is not as easy as it may seem. 

As this PR goes against CMS policy **not to change existing xml files** it should be discussed at Sim meeting when @cvuosalo is back. Also, who will take responsibility to create and validate the new payloads should be discussed and agreed. The change in **materials.xml** file touches all of them. The GTs will have to be updated in order to insure that the next significant change in Geometry is unrelated to the materials restructuring.

**Review by ianna — CHANGES_REQUESTED** [review]
@bsunanda, please, check the comments and cleanup the xml files and their scenarios. Most of the material variation definitions were produced automatically. Modifying them by hand is not as easy as it may seem. 

As this PR goes against CMS policy **not to change existing xml files** it should be discussed at Sim meeting when @cvuosalo is back. Also, who will take responsibility to create and validate the new payloads should be discussed and agreed. The change in **materials.xml** file touches all of them. The GTs will have to be updated in order to insure that the next significant change in Geometry is unrelated to the materials restructuring.

**Comment by silviodonato** [comment]
@bsunanda Do you have news about this PR?

**Comment by bsunanda** [comment]
@silviodonato I shall take comments from Yana, implement them and revive this PR.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by cvuosalo** [comment]
hold

**Comment by cvuosalo** [comment]
@bsunanda: This PR needs to be discussed at the next Sim meeting, so it should stay on hold until then. Please make a presentation at the next Sim meeting explaining how the payloads and GTs will be updated and by whom, as Yana requested.

**Comment by silviodonato** [comment]
@bsunanda @cvuosalo could you write a comment about the outcome about this PR of the Sim meeting? thanks

**Comment by cvuosalo** [comment]
I will try to do a test to assess the impact of this PR on the payloads.

File: Geometry/TrackerCommonData/data/2016/FlatMinus05Percent/trackermaterial.xml
Comment (cvuosalo) [inline Geometry/TrackerCommonData/data/2016/FlatMinus05Percent/trackermaterial.xml]: @bsunanda, @ianna There are two "micro_twisted_power_cable" composite materials defined here. Aren't duplicate names an error? What effect does this duplication have? Does anyone in CMS know?

**Comment by cvuosalo** [comment]
@bsunanda, @ianna, @civanch I keep running into show-stoppers in attempting to validate this PR. We may need to have a meeting to develop a clear, workable protocol for validation.

**Comment by silviodonato** [comment]
Do you have any news on this PR?

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@silviodonato @cvuosalo @ianna @civanch I tried to resolve duplicate names which came from original files

**Comment by silviodonato** [comment]
Do you have any comments about this PR? @Dr15Jones @civanch  @cvuosalo @ianna @makortel @mdhildreth

**Comment by bsunanda** [comment]
As far as I see, this is okay.



Sent from my Verizon 4G LTE smartphone


-------- Original message --------

**Comment by cvuosalo** [comment]
Validation of these XML changes continues to be very difficult. There does not seem to be a workable process yet to check the XML, payloads, and database. Discussions are ongoing.

**Comment by cvuosalo** [comment]
@bsunanda One conclusion of discussions with Yana is that XML files for Run 2 and earlier should not be changed. The validation process for changing any XML is already quite onerous, but to do so for all the old scenarios verges on being unfeasible. Also, it is not clear that processing old scenarios with DD4hep will ever be even necessary.

Please remove files for old scenarios from this PR. Only the XML used for 2021 and later and Run 3 and later should be included in this PR. Thanks.

**Comment by cvuosalo** [comment]
The timeline for this PR will have to be significantly extended. We have discovered significant discrepancies between geometry XML and the DB payloads in the current master. Before any geometry XML files can be changed, new payloads and global tags will need to be generated from the current master and validated to establish a reliable baseline. This process may be quite lengthy. Only after we have established the baseline can new payloads and global tags from this PR be tested to see whether they cause any differences.

**Comment by cvuosalo** [comment]
unhold

**Comment by cvuosalo** [comment]
This PR can probably go ahead. Comparison with the Conditions DB shows that there is no real change in simulation geometry.

**Comment by cvuosalo** [comment]
I am not sure how meaningful it is to change the files for Run 2 and earlier scenarios. New payloads for these scenarios are very unlikely to be created with CMSSW 11_1 or later versions. My impression is the the Ultra-Legacy is the final word for Run 2.

**Comment by silviodonato** [comment]
please test
refresh

**Comment by silviodonato** [comment]
code-checks

**Comment by bsunanda** [comment]
@cvuosalo @kpedro88 Can you sign this now?

**Comment by cvuosalo** [comment]
I have tested more carefully. The Run 1 payloads from this PR don't seem to match what is in Conditions DB. Do we really want to change Runs 1-2 right now? I am unsure how we want to approach DD4hep migration for Runs 1-2 at this time.

**Comment by fabiocos** [comment]
@cvuosalo could you please elaborate on the differences observed in your test?

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
hold

**Comment by cvuosalo** [comment]
The conclusion that is emerging is that we don't want to change Runs 1-2 XML files. Ultimately, they may be deleted. I was hoping to discuss this issue at the planned April 3rd Sim meeting, but that meeting has been postponed until April 10th.
In the interim, I suggest closing this PR. A new PR with files only for 2021 and later could then be submitted so we could go ahead with those changes. After the April 10th meeting, we may have a final decision on what to do with Runs 1-2 geometry XML files.

**Comment by cvuosalo** [comment]
@fabiocos There are several differences between Run 1 XML and the geometry in the DB. I suppose these differences can be explained by inadvertent changes being made to the Run 1 XML over the years while the DB geometry has not been touched.

There are differences in the concatenated geometry XML file from CMSSW 11_1 and the one in the DB. For example:
```
25987c25945
< <Box dx="138.500000000000000000*mm" dy="15.349999999999999645*mm" dz="4.650000000000000355*mm" name="esalgo:SFAbs1PbLOutAltmp"/>
---
> <Box dx="1384.099999999999909051*mm" dy="15.349999999999999645*mm" dz="4.650000000000000355*mm" name="esalgo:SFAbs1PbLOutAltmp"/>
26094c26052
< <Box dx="169.800000000000011369*mm" dy="20.350000000000001421*mm" dz="1.850000000000000089*mm" name="esalgo:SFAbs2PbLOutAltmp"/>
---
> <Box dx="1697.099999999999909051*mm" dy="20.350000000000001421*mm" dz="1.850000000000000089*mm" name="esalgo:SFAbs2PbLOutAltmp"/>
26254c26212
< <Box dx="20.149999999999955946*mm" dy="158.550000000000011369*mm" dz="1.850000000000000089*mm" name="esalgo:SFAbs2PbLOutAltmp2"/>
---
> <Box dx="20.149999999999955946*mm" dy="1584.599999999999909051*mm" dz="1.850000000000000089*mm" name="esalgo:SFAbs2PbLOutAltmp2"/>
26309c26267
< <Box dx="24.150000000000069633*mm" dy="173.400000000000005684*mm" dz="4.650000000000000355*mm" name="esalgo:SFAbs1PbLOutAltmp2"/>
---
> <Box dx="24.150000000000069633*mm" dy="1733.099999999999909051*mm" dz="4.650000000000000355*mm" name="esalgo:SFAbs1PbLOutAltmp2"/>
```
If you look carefully, you can see some of these lengths changed by a factor of 10.

The Simulation geometry also has differences.

I think all these Run 1 differences are moot because I don't think we have any plan to create new Run 1 DB payloads.

**Comment by fabiocos** [comment]
@cvuosalo I see, so this has nothing to do with the material reorganization. In any case a factor of 10 in preshower volumes does not sound healthy, but it depends on the real use of these volumes. And I do not see them in Run2/Run3 xml geometry.

In any case, if we update xmls we have unrelated changes already, and if we want to keep the capability to produce Run1 geometries when DDD has gone I think we need to make the update, right? Although I am not sure this update is specifically needed by DD4hep (@bsunanda could you please confirm?)

**Comment by civanch** [comment]
Hi all,

because we do not want change DB geometry for Run-1,2 can we sign this PR, which for today will change XML geometry for Run-3 and Phase-2 any. Any result for Run-1,2 will not be changed.

We agreed to restore functionality of XML for Run1,2, when Run3 will be completed and DD4Hep migration will be done. The condition: XML for Run1,2 should be identical to DB Run1,2. Only after that we can drop old DD.

@ianna , @cvuosalo , would you agree?

**Comment by cvuosalo** [comment]
@civanch What I understand you saying is that, since we are not planning to make new DB payloads for Runs 1-2 in the near future, we are then free to change the Runs 1-2 XML files. The problem of the Runs 1-2 XML files will be pushed back until after the DD4hep migration is completed.

But the plan you mentioned concerns me. The stated condition "XML for Runs 1-2 should be identical to the DB payloads for Runs 1-2" is true for the legacy releases of CMSSW (5_3 and 10_6) but will pose significant problems for new versions of CMSSW. We will need to discuss it at the next Sim meeting.

**Comment by cvuosalo** [comment]
unhold

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
hold

> We will need to discuss it at the next Sim meeting.

**Comment by silviodonato** [comment]
If I understand correctly, 
- the current XML geometry does not match the DB geometry
- this PR does not change the XML geometry
- hence, even after this PR, we will continue to have a XML geometry that does not match the DB geometry

If this is correct, I think we can safety merge this PR.
The mismatch between XML geometry and DB geometry is a different issue and will be fixed with another PR.

**Comment by ianna** [comment]
@silviodonato - you are correct.

I'd suggest to merge this PR and upload the new DB payloads for all Runs (and update the Run3 GTs). The GT change will result in a baseline comparison differences. However, we know that it would be superficial at this point. Then we'd need to update the baseline. This would make sure we'd catch any further geometry changes resulting in a baseline comparison differences.

**Comment by civanch** [comment]
I agree with Ianna. Probably, optimal would be to complete fixing of Run-3 geometry and after make GTs for Run-3? We may agree, at what pre-release we stop updating Run-3 geometry. In a mean time this PR may be merged.

**Comment by silviodonato** [comment]
unhold
ok, I will integrate this PR in the next IB

**Comment by silviodonato** [comment]
+1

**Pull request merged**
